awx/main/scheduler/task_manager.py

```diff
@@ -132,13 +135,19 @@ def schedule(self):
                         return
                     logger.debug(f"Starting {self.prefix} Scheduler")
                     # if sigterm due to timeout, still record metrics
-                    signal.signal(signal.SIGTERM, self.record_aggregate_metrics_and_exit)
+                    signal.signal(signal.SIGUSR1, self.record_aggregate_metrics_and_exit)
                     self._schedule()
                     commit_start = time.time()
 
                 if self.prefix == "task_manager":
                     self.subsystem_metrics.set(f"{self.prefix}_commit_seconds", time.time() - commit_start)
                 self.record_aggregate_metrics()
+
+                # Reset the signal handler back to the default just in case anything
+                # else uses the same signal for other purposes
+                logger.debug('Resetting SIGUSR1 handler')
+                signal.signal(signal.SIGUSR1, original_sigusr1)
```

Comment:
I'd like to see this in a `finally:`, but I can't tell you the exact structure that should have.